pkg/cmd/fixcerts/fixcerts.go

```diff
@@ -0,0 +1,407 @@
+	kubeapiservercertrotationcontroller "github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/certrotationcontroller"
+	kubeapiserveroperatorclient "github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/operatorclient"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/recovery"
+)
+
+const (
+	KubeControllerManagerStaticPodFileName = "kube-controller-manager-pod.yaml"
+	KubeSchedulerStaticPodFileName         = "kube-scheduler-pod.yaml"
+
+	secretsType    = "secrets"
+	configmapsType = "configmaps"
+)
+
+var (
+	Scheme = runtime.NewScheme()
+	Codecs = serializer.NewCodecFactory(Scheme)
+)
+
+func init() {
+	metav1.AddToGroupVersion(Scheme, schema.GroupVersion{Version: "v1"})
+	utilruntime.Must(corev1.AddToScheme(Scheme))
+}
+
+type Options struct {
+	// TODO: merge with CreateOptions
+	KubeApiserverImage    string
+	PodManifestDir        string
+	StaticPodResourcesDir string
+	Timeout               time.Duration
+}
+
+func NewCommand() *cobra.Command {
+	o := &Options{
+		KubeApiserverImage:    "", // TODO: set the public pullspec
+		PodManifestDir:        "/etc/kubernetes/manifests",
+		StaticPodResourcesDir: "/etc/kubernetes/static-pod-resources",
+		Timeout:               5 * time.Minute,
+	}
+
+	cmd := &cobra.Command{
+		Use: "fix-certs",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := o.Complete()
+			if err != nil {
+				return err
+			}
+
+			err = o.Validate()
+			if err != nil {
+				return err
+			}
+
+			err = o.Run()
+			if err != nil {
+				return err
+			}
+
+			return nil
+		},
+
+		SilenceErrors: true,
+		SilenceUsage:  true,
+	}
+
+	// TODO: add/prettify descriptions
+	cmd.Flags().StringVar(&o.KubeApiserverImage, "kube-apiserver-image", o.KubeApiserverImage, "")
+	cmd.Flags().StringVar(&o.StaticPodResourcesDir, "static-pod-resources", o.StaticPodResourcesDir, "path to store a directory containing static pod resources for recovery apiserver manifest")
+	cmd.Flags().StringVar(&o.PodManifestDir, "pod-manifest-dir", o.PodManifestDir, "directory for the static pod manifes")
+	cmd.Flags().DurationVar(&o.Timeout, "timeout", o.Timeout, "timeout, 0 means infinite")
+
+	return cmd
+}
+
+func (o *Options) Complete() error {
+	return nil
+}
+
+func (o *Options) Validate() error {
+	if o.KubeApiserverImage == "" {
+		return errors.New("missing required flag --kube-apiserver-image")
+	}
+
+	if o.PodManifestDir == "" {
+		return errors.New("pod-manifest-dir shall not be empty")
+	}
+
+	return nil
+}
+
+func (o *Options) Run() error {
+	ctx, cancel := watch.ContextWithOptionalTimeout(context.Background(), o.Timeout)
+	defer cancel()
+	// TODO: hook up signals
+
+	apiserver := &recovery.Apiserver{
+		KubeApiserverImage:    o.KubeApiserverImage,
+		PodManifestDir:        o.PodManifestDir,
+		StaticPodResourcesDir: o.StaticPodResourcesDir,
+	}
+
+	err := apiserver.Create()
```

Comment:
We should not do this.  Fixing the certs should use the `admin.kubeconfig` from it's known location.  It should not try to start a kube-apiserver